pkg/scm/git/git.go

```diff
@@ -62,18 +68,107 @@ func stringInSlice(s string, slice []string) bool {
 // ValidCloneSpec determines if the given string reference points to a valid git
 // repository
 func (h *stiGit) ValidCloneSpec(source string) bool {
-	url, err := url.Parse(source)
-	if err != nil {
+	return validCloneSpec(source, false, nil)
+}
+
+// ValidCloneSpecRemoteOnly determines if the given string reference points to a valid remote git
+// repository; valid local repository specs will result in a return of false
+func (h *stiGit) ValidCloneSpecRemoteOnly(source string) bool {
+	return validCloneSpec(source, true, nil)
+}
+
+// MungeNoProtocolURL will take a URL returned from net/url.Parse and make
+// corrections to the URL when no protocol is specified in the Scheme
+func (h *stiGit) MungeNoProtocolURL(source string, url *url.URL) bool {
+	return validCloneSpec(source, false, url)
+}
+
+func validCloneSpec(source string, remoteOnly bool, partialURL *url.URL) bool {
+	if partialURL == nil {
+		uri, err := url.Parse(source)
+		if err != nil {
+			return false
+		}
+		partialURL = uri
+	}
+
+	if strings.Index(source, "~") != -1 {
```

Comment:
Yep ... it works fine now with the latest regex's and that check removed.  I'll update the PR.